From 5ac7eda8ab013492ae4c67c7b581a72ae8960106 Mon Sep 17 00:00:00 2001
Message-Id: <5ac7eda8ab013492ae4c67c7b581a72ae8960106.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Shashank Sharma <shashank.sharma@intel.com>
Date: Wed, 28 Jan 2015 13:13:53 +0530
Subject: [PATCH 23/95] MUST_REBASE [VPG]: adf/intel/vlv: Add encoder
 parameters

This patch creates field inside intel_pipeline that can be used by
individual encoders to store values to be passed to platform code. The
purpose of this is to have a common location to transmit data between
encoders and platform code.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5478
Change-Id: Ia31adbffa6057abe056dcba6abb993a22772436d
Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
---
 .../video/adf/intel/include/core/intel_dc_config.h |   29 +++++++++++++-------
 1 file changed, 19 insertions(+), 10 deletions(-)

diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index e819f6c..0d0fad2 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -489,17 +489,26 @@ struct intel_dc_attachment {
 	u8 pll_id;
 };
 
+union encoder_params {
+	/*
+	 * pass display specific params to pipeline code
+	 * that might be needed across multiple calls
+	 */
+	struct {
+		struct dsi_config *dsi_config;
+	} dsi;
+
+	struct {
+		int temp;/*stub*/
+	} dp;
+
+	struct {
+		u8 bpp;
+	} hdmi;
+};
+
 struct intel_pipeline {
-	union {
-		/* pass display specific params to pipeline code */
-		/* that might be needed across multiple calls */
-		struct {
-			struct dsi_config *dsi_config;
-		} dsi;
-		struct {
-			int temp;/*stub*/
-		} dp;
-	} params;
+	union encoder_params params;
 };
 
 struct intel_dc_config {
-- 
1.7.9.5

